src/dom-manipulation.ts

```diff
@@ -0,0 +1,293 @@
+  const selectedFlightInfo: IFlight | undefined = flightArray.find((flight: { icao24: string; }) => flight.icao24 === selectedFlight)
+  if(!selectedFlightInfo) return;
+  const selectedFlightInfoRow = shadowRoot.getElementById(
+    selectedFlightInfo.icao24
+  ) as HTMLElement;
+
+  const verticalRateElement = selectedFlightInfoRow.querySelector(".vertical-rate");
+  const velocityElement = selectedFlightInfoRow.querySelector(".velocity");
+  const directionElement = selectedFlightInfoRow.querySelector(".direction");
+  const headingElement = selectedFlightInfoRow.querySelector(".heading");
+  const altitudeElement = selectedFlightInfoRow.querySelector(".altitude");
+
+  if (!verticalRateElement || !velocityElement || !directionElement || !headingElement || !altitudeElement) return;
+  verticalRateElement.innerHTML = `${convertMeterPerSecondToKilomentersPerHour(
+    selectedFlightInfo.vertical_rate
+  )}km/h`
+  velocityElement.innerHTML = `${convertMeterPerSecondToKilomentersPerHour(
+    selectedFlightInfo.velocity
+  )}km/h`
+  directionElement.innerHTML = `${calculateDirection(
+    selectedFlightInfo.true_track
+  )}`
+  headingElement.innerHTML = `${selectedFlightInfo.true_track}°`
+  altitudeElement.innerHTML = `${selectedFlightInfo.baro_altitude}m`
+  
+  setMapAndMarkerToCurrentFlightLocation(
+    selectedFlightInfo.latitude,
+    selectedFlightInfo.longitude,
+    selectedFlightInfo.true_track
+  );
+}
+
+
+
+function upsertFlightListData(flightArray: IFlight[], shadowRoot: ShadowRoot) {
+  flightArray.map((flight) => {
+    const exsitingFlightInfoRow = shadowRoot.getElementById(
+      flight.icao24
+    ) as HTMLElement;
+    const flightButton = shadowRoot.getElementById(
+      flight.icao24 + flight.callsign
+    ) as HTMLElement;
+
+    if (!shadowRoot) return;
+    if (!flightButton) {
+      shadowRoot.innerHTML += createNewFlightInfoRow(flight);
+    } else {
+      exsitingFlightInfoRow.innerHTML = appendExistingFlightInfoRow(flight);
+    }
+  });
+  minimiseLoadingScreen();
+  addEventListenerToFlightInfoButtons(flightArray, shadowRoot);
+}
+
+function cleanUpFlightListData(flightArray: IFlight[], shadowRoot: ShadowRoot) {
+  const inScopeFlightCodes: string[] = [];
+  const flightInfoDivs = shadowRoot.querySelectorAll(".single-flight");
+
+  flightArray.map((flight) => inScopeFlightCodes.push(flight.icao24));
+  flightInfoDivs.forEach((flightRow) => {
+    const inScope = inScopeFlightCodes.includes(flightRow.id);
+    if (!inScope) {
+      const singleFlightToRemove = shadowRoot.getElementById(flightRow.id);
+      singleFlightToRemove?.remove();
+    }
+  });
+}
+
+function createNewFlightInfoRow(flight: IFlight): string {
+  if (!flight.icao24) return "";
+  return `
+  <div id="${
+    flight.icao24
+  }" class="single-flight">
+      <span class="callsign">${flight.callsign}</span>
+      <span class="vertical-rate">${convertMeterPerSecondToKilomentersPerHour(
```

Comment:
did you consider using `p` tags since you are displaying text?

Reply:
I used span as it placed the text next to each other on initial testing.

But that was made redundant by grid.